src/net.cpp

```diff
@@ -93,7 +93,13 @@ void CConnman::AddOneShot(const std::string& strDest)
 
 unsigned short GetListenPort()
 {
-    return (unsigned short)(gArgs.GetArg("-port", Params().GetDefaultPort()));
+    int default_port = Params().GetDefaultPort();
+    assert((0 < default_port) && (default_port <= UINT16_MAX));
+    // An int that fits in uint16 always fits in int64_t
+    int64_t listen_port = gArgs.GetArg("-port", default_port);
+    // We don't support the reserved port 0 for system-allocated dynamic ports yet.
+    assert((0 < listen_port) && (listen_port <= UINT16_MAX));
```

Comment:
In commit "tor: pass in Tor control port rather than using globals" (7e314526e7c4bb5bbf0e351f92cab940158eae24)

It's not a good idea to validate user input with an assert, because it causes the program to crash suddenly and different environments to display the error in confusing ways (in stderr, system logs, or popup windows, without the full context).

Simplest thing to do here would be to drop this assert, change the function return value to int64_t, and restore the error checking code in `init.cpp` where this function is called:

```c++
if ((listen_port <= 0) || (0xffff < listen_port)) {
    return InitError(strprintf(_("Invalid port specified in -port: '%i'"), listen_port));
}
```

Keeping the other assert for `default_port` should be ok, since that value is compiled into the program, and if it is out of range it indicates there's probably a bug.